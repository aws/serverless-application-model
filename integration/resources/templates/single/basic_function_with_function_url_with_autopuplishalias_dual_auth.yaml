Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs18.x
      CodeUri: ${codeuri}
      MemorySize: 128
      AutoPublishAlias: live
      FunctionUrlConfig:
        AuthType: NONE
        Cors:
          AllowOrigins:
          - https://foo.com
          AllowMethods:
          - POST
          AllowCredentials: true
          AllowHeaders:
          - x-Custom-Header
          ExposeHeaders:
          - x-amzn-header
          MaxAge: 10
Outputs:
  FunctionUrl:
    Description: URL of the Lambda function alias
    Value: !GetAtt MyLambdaFunctionUrl.FunctionUrl
Metadata:
  SamTransformTest: true
