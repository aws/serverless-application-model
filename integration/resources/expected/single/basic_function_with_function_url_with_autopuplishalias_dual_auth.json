[
  {
    "LogicalResourceId": "MyLambdaFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionVersion",
    "ResourceType": "AWS::Lambda::Version"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionAliaslive",
    "ResourceType": "AWS::Lambda::Alias"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionUrlPublicPermissions",
    "ResourceType": "AWS::Lambda::Permission"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionURLInvokeAllowPublicAccess",
    "ResourceType": "AWS::Lambda::Permission"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionUrl",
    "ResourceType": "AWS::Lambda::Url"
  }
]
